src/Integration.UnitTests/Integration.UnitTests.csproj

```diff
@@ -7,6 +7,10 @@
     <RequiresRuleset>true</RequiresRuleset>
     <RequiresAlmAuthentication>true</RequiresAlmAuthentication>
     <RequiresNewtonsoft>true</RequiresNewtonsoft>
+    <!-- Conditional compilation constant for VS2019-specific code.
+         We'd prefer to set this the shared SonarLint.VSSpecificAssemblies.props file, but
+         the IDE isn't picking up the setting if we do that. -->
+    <DefineConstants Condition="($(VisualStudioVersion)=='17.0' AND $(BuildingInsideVisualStudio)=='true') OR $(VSTargetVersion)=='2022'">$(DefineConstants);VS2022</DefineConstants>
```

Comment:
I would hope that fixes I made in the target files to ensure VSTargetVersion is set correctly in the IDE means the first part of this condition isn't strictly necessary, but there's no harm in having it.